### ISSUE TITLE
 Why \u0120 (Ġ) is in many pairs?

### ISSUE THREAD

**Issue by fbparis** [openai/gpt-2#80]
It seems obvious that it's used as a "blank byte" but I don't understand how this "blank byte" ends up being represented by this "Ġ". Can someone explain this?

**Comment by lostmsu** [comment]
I am not with OpenAI, but my understanding is that this character simply indicates the beginning of a word. I think its presence improves model's behavior around word boundaries.

As for technical details, it is simply an encoded space character. This version of GPT-2 seem to be using the space character itself for other purposes.

**Comment by 8enmann** [comment]
The encoder code doesn't like spaces, so they replace spaces and other whitespace characters with other unicode bytes. See `encoder.py` for details.

**Comment by kamatsuoka** [comment]
That's right, the bytes_to_unicode function takes all control and whitespace characters in code points 0-255 and shifts them up by 256 to make them printable.  So space (code point 32) becomes Ġ (code point 288).

 https://github.com/openai/gpt-2/blob/master/src/encoder.py#L9
